wasmer/import_object.go

```diff
@@ -29,11 +32,18 @@ func (self *ImportObject) intoInner() *C.wasm_extern_vec_t {
 	var externs []*C.wasm_extern_t
 	var numberOfExterns uint
 
-	for _, namespace := range self.externs {
-		for _, extern := range namespace {
-			externs = append(externs, extern.IntoExtern().inner())
-			numberOfExterns++
+	for _, importType := range module.Imports() {
+		namespace := importType.Module()
+		name := importType.Name()
+
+		if self.externs[namespace][name] == nil {
+			return nil, &Error{
+				message: fmt.Sprintf("Missing import: `%s`.`%s`", namespace, name),
+			}
 		}
+
+		externs = append(externs, self.externs[namespace][name].IntoExtern().inner())
+		numberOfExterns++
 	}
 
 	for _, extern := range self.opaqueExterns {
```

Comment:
@Hywan what are these `opaqueExterns` for? How can we fix them?

They are likely to have the exact same problem as "normal" externs.

Reply:
It's an artifact to support WASI with `ImportObject` as far as I remember. Let's keep it for the moment.